pkg/engine2/operational_eval/vertex_path_expand.go

```diff
@@ -471,15 +472,8 @@ func (v *pathExpandVertex) Dependencies(eval *Evaluator) (graphChanges, error) {
 	changes := newChanges()
 	srcKey := v.Key()
 
-	var errs error
-	if propertyReferenceInfluencesEdge(v.Satisfication.Source) {
-		keys := getDepsForPropertyRef(eval.Solution, v.Edge.Source, v.Satisfication.Source.PropertyReference)
-		changes.addEdges(srcKey, keys)
-	}
-	if propertyReferenceInfluencesEdge(v.Satisfication.Target) {
-		keys := getDepsForPropertyRef(eval.Solution, v.Edge.Target, v.Satisfication.Target.PropertyReference)
-		changes.addEdges(srcKey, keys)
-	}
+	changes.addEdges(srcKey, getDepsForPropertyRef(eval.Solution, v.Edge.Source, v.Satisfication.Source.PropertyReference))
```

Comment:
This was 1 part of the fix - the dependencies were getting skipped due to it having Validity.